daemon/api.go

```diff
@@ -673,6 +674,34 @@ func newChange(st *state.State, kind, summary string, tsets []*state.TaskSet) *s
 
 const maxReadBuflen = 1024 * 1024
 
+func trySnap(c *Command, r *http.Request, user *auth.UserState, trydir string) Response {
+	st := c.d.overlord.State()
+	st.Lock()
+	defer st.Unlock()
+
+	msg := fmt.Sprintf(i18n.G("Try %q snap"), trydir)
+
+	if !osutil.IsDirectory(trydir) {
+		return InternalError("cannot try %q not a directory", trydir)
+	}
+
+	info, err := readSnapInfo(trydir)
+	if err != nil {
+		return BadRequest("cannot read snap info for %s: %s", trydir, err)
+	}
+
+	tsets, err := snapstateInstallPath(st, info.Name(), trydir, "unused-channel", 0)
```

Comment:
"unused-channel" feels strange.. I think this should be just "", same as we do for snap files.